app/master/build.py

```diff
@@ -99,7 +110,8 @@ def prepare(self, subjobs, job_config):
         :type subjobs: list[Subjob]
         :type job_config: JobConfig
         """
-        if self.project_type is None:
+        self._record_state_timestamp(BuildStatus.PREPARING)
```

Comment:
Ha! Good catch -- I definitely meant to include the fetch time. That's more complicated then; I think I'll just remove `PREPARING` for now and save that for a future change.